### ISSUE TITLE
Bug in save model

### ISSUE THREAD

**Issue by faneshion** [NTMC-Community/MatchZoo#726]
Bug exists when saving the model with the following setting:
```
model.params['optimizer'] = keras.optimizers.Adam(lr=0.001)
```
The callback is :
```
evaluate = mz.callbacks.EvaluateAllMetrics(model, x=test_x, y=test_y, batch_size=len(test_x), model_save_path='./.tmpdir/')
```

**Comment by sth4k** [comment]
you can just use model.save('./.tmpdir/') and load it with from matchzoo.engine.base_model import load_model

**Comment by presimple** [comment]
> `params` keys should only be basic data types (string, int), not keras optimizer instances. However, I know how to make this work (by calling keras seriailization helpers). Do you think it's really needed?

Hi, I found this when I tried to use specific learning rate to train my model. Or could you give a better way to specify a customized learning rate with the optimizer while using matchzoo?

**Comment by uduse** [comment]
@presimple 
If you want a quick solution:
1. add an additional parameter `learning_rate` in `model.params` and modify `model.build` so it takes that parameter into account
2. add an additional parameter `learning_rate` in `model.params` and add additional logic to read that parameter and compile a loaded model.

A better solution, but takes longer:
Matchzoo handles optimizer serialization differently.

Ultimately it's [keras's fault](https://github.com/keras-team/keras/issues/789). Maybe they will fix that eventually.

**Comment by uduse** [comment]
@presimple found a much better workaround in #735

**Comment by uduse** [comment]
I hope things are working well for you now. I’ll go ahead and close this issue, but I’m happy to continue further discussion whenever needed.
